mocks/mocks.py

```diff
@@ -679,6 +681,12 @@ def convert_to_lumonsity_distance(particle_positions,
     astropy.units.Quantity
         The data with converted to luminosity distances along viewing axis.
     """
+    try:
+        particle_positions.dtype = 'float64'
```

Comment:
This is _not_ the right way to change the data type of a numpy array. Instead of assigning to the `.dtype` attribute you should use the `.astype()` method.  Also, this is the wrong place to be doing data type checks/conversions. That should be in the function that loads the data in the first place.

Reply:
I made changes and moved the block to the `scale_light_by_distance` function.

Reply:
That's still the wrong place for it! Converting the data type of the simulation data has nothing to do scaling light by distance, so it shouldn't be in a function that does that. It should be in a function that loads & parses the simulation data, as I also mention in the other PR: https://github.com/AstroHuntsman/huntsman-mocks/pull/69#discussion_r276055664

Also, you still have the incorrect line `particle_positions.dtype = 'float64'` where you assign to `.dtype`, which you shoudn't do, and the whole block is unnecessary anyway.

All you need is a single line, `particle_positions = particle_positions.astype(np.float64)`.  That will convert the particle position data to 64 bit floats if its not already, or do nothing if it's already 64 bits. You don't need a warning if the `particle_position` is 32 bit floats to begin with because that's expected for at least some of the data you expect this code to be used with (GADGET-2 output).

Reply:
Will be solved due to issue #74

Reply:
If you want to do that as a separate PR then I'd suggest doing it before continuing with the two PRs currently under review. Otherwise you'll be spending time writing code that you change again almost immediately. #74 is mostly a bit of copy-pasting, it's not a big change.

Reply:
Alright, so I will do it in #69 and delete this PR and issue #74 .

Reply:
You can leave Issue #74, just close it when fix it in PR #69. This PR only needs to be a single line of code, as noted above, but that single line is still needed. Can be done in this PR.

Reply:
So you make it a single line where it is now? because I am going to create the `config_parse` function in #69 .

Reply:
In the data loading function, as I said here: https://github.com/AstroHuntsman/huntsman-mocks/pull/69#discussion_r276055664